File: Backend/LibrarySystem.Models/Models/Book.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LibrarySystem.Models.Models
{
    public class Book
    {
        public int Id { get; set; }
        public string? Title { get; set; }
        public string? ISBN { get; set; }
        public int PageCount { get; set; }
        public int PublicationYear { get; set; } 
        public string? Language { get; set; }
        public string? ImageUrl { get; set; }
        public string? Summary { get; set; }
        public int CategoryId { get; set; }
        public int PublisherId { get; set; }
        public Category? Category { get; set; }
        public Publisher? Publisher { get; set; }
        public ICollection<BookCopy> BookCopies { get; set; } = new List<BookCopy>();
        public ICollection<BookAuthor> BookAuthors { get; set; } = new List<BookAuthor>();
        public ICollection<BookComment> Comments { get; set; } = new List<BookComment>();
    }
}

File: Frontend/src/types/book.ts
import {BookAuthor, BookCopy} from '@/src/types/bookDetail'
import { Publisher } from '@/src/types/publisherAndAuthor'
import {Category} from '@/src/types/category'

export interface Book {
    id: number;
    title: string;
    isbn: string;
    pageCount: number;
    publicationYear: number;
    language: string;
    category: Category;
    publisher: Publisher;
    bookAuthors: BookAuthor[];
    bookCopies: BookCopy[];
    authorId: number;
    authorFirstName: string;
    authorLastName: string;
    categoryId: number;
    categoryName: string;
    publisherId: number;
    publisherName: string;
    imageUrl: string;
    summary: string;
}


export interface PaginatedResult<T> {
    items: T[];
    totalCount: number;
    page: number;
    pageSize: number;
    totalPages: number;
}

export interface BookFilterDto {
    page?: number;
    size?: number;
    title?: string;
    categoryId?: number;
    authorId?: number;
    publisherId?: number;
    publicationYearFrom?: number;
    publicationYearTo?: number;
    language?: string;
    pageCountMin?: number;
    pageCountMax?: number;
    hasAvailableCopy?: boolean;
    roomCode?: string;
}

export interface CopyFilterDto {
    bookId: number;
    page: number;
    size: number;
    sortBy?: string; // 'shelfCode', 'roomCode', 'barcode'
    sortOrder?: 'asc' | 'desc';
}

export interface CreateBookResponseDto{
    id: number;
}

export interface CreateBookCopyResponseDto{
    id: number;
}

export interface CreateCopyBookDto{
    bookId: number;
    shelfCode: string;
    barcodeNumber: string;
    roomId: number;
}

export interface CreateBookDto {
    title?: string;
    isbn?: string;
    pageCount: number;
    publicationYear: number;
    language?: string;
    authorId?: number;
    authorFirstName?: string;
    authorLastName?: string;
    categoryId?: number;
    categoryName?: string;
    publisherId?: number;
    publisherName?: string;
    imageUrl?: string;
    summary?: string;
}

export interface UpdateBookDto {
    id: number;
    title: string;
    isbn: string;
    pageCount: number;
    publicationYear: number;
    language: string;
    authorId: number;
    publisherId: number;
    categoryId: number;
    imageUrl: string;
    summary: string;
}

export interface BookComment {
    id: number;
    userName: string; // Veya User nesnesi
    content: string;
    rating: number; // 1-5 arası
    createdAt: string;
}